punq/__init__.py

```diff
@@ -114,23 +114,23 @@ def build_context(self, key, existing=None):
 
         return existing
 
-    def register(self, service, _factory=None, **kwargs):
+    def register(self, service, factory=None, instance=None, **kwargs):
         resolve_args = kwargs or {}
 
-        if _factory is None:
+        if instance is not None:
```

Comment:
Are there any reasonable use-cases where people might want to inject `None` as instance? Maybe we can use `_Sentinel()` as a default?

Reply:
I don't think there are any *reasonable* use cases for it, but people do unreasonable things. Added explicit `sentinel` value to the punq module.